src/js/new-visualization-modal.js

```diff
@@ -54,7 +47,54 @@ export default function initModal() {
   // add SQL query to the modal
   document.getElementById('data-query').innerHTML = dataQuery;
 
+  hideAllWarnings();
+
   // wire up event listeners
   $('#new-vis-button').click(onNewVisSubmit);
   $('#copy-query-button').click(onCopyQueryClick);
 }
+
+const csvStringToArray = (strData) => {
+  const objPattern = new RegExp('(\\,|\\r?\\n|\\r|^)(?:"([^"]*(?:""[^"]*)*)"|([^\\,\\r\\n]*))', 'gi');
+  // Create an array to hold our individual pattern
+  // matching groups.
+  let arrMatches = null,
+    // Create an array to hold our data. Give the array
+    // a default empty first row.
+    arrData = [[]];
+  // Keep looping over the regular expression matches
+  // until we can no longer find a match.
+  while ((arrMatches = objPattern.exec(strData))) {
+    if (arrMatches[1].length && arrMatches[1] !== '\\') arrData.push([]);
+    arrData[arrData.length - 1].push(arrMatches[2] ? arrMatches[2].replace(new RegExp('""', 'g'), '"') : arrMatches[3]);
+  }
+  return arrData;
+};
+
+function readSingleFile(evt) {
+  let f = evt.target.files[0];
+  if (f) {
+    let r = new FileReader();
+    r.onload = function (e) {
+      let contents = e.target.result;
+      let lines = csvStringToArray(contents);
+      lines.shift();
+
+      try {
+        queryResults = lines.map((line) => {
+          return JSON.parse(line);
+        });
+        hideAllWarnings();
+      } catch (error) {
+        $('#invalid-query-results-warning').show();
+        alert(
+          "Whoops! We couldn't parse your results. Please double check your query and try again. Please submit feedback if you believe this is a bug."
+        );
+      }
+    };
+    r.readAsText(f);
+  } else {
+    alert('Failed to load file');
```

Comment:
Do not use alerts. Find a better way to handle this issue.

Reply:
Alerts removed